src/seedsigner/views/seed_views.py

```diff
@@ -2294,3 +2301,289 @@ def run(self):
 
         # Exiting/Canceling the QR display screen always returns Home
         return Destination(MainMenuView, skip_current_view=True)
+
+
+
+"""****************************************************************************
+    Shamir's Secret Sharing Views
+****************************************************************************"""
+class SeedEntryShamirThresholdView(View):
+    def __init__(self):
+        super().__init__()
+
+
+    def run(self):
+        title = _("SLIP-39 Threshold")
+        ret_dict = self.run_screen(seed_screens.SeedEntryShamirThresholdScreen, entered_number="", title=title)
+
+        if "is_back_button" in ret_dict:
+            return Destination(BackStackView)
+            
+        elif ret_dict["entered_number"] != "" and ret_dict["entered_number"] != "0":
+            return Destination(SeedShamirShareImportSelectWordCount, view_args={"k": int(ret_dict["entered_number"])})
+
+        else:
+            if ret_dict["entered_number"] == "0":
+                from seedsigner.gui.toast import ErrorToast
+                self.controller.activate_toast(ErrorToast(_("Threshold not valid")))
+
+        return Destination(BackStackView)
+        
+
+
+class SeedShamirShareImportSelectWordCount(View):
+    TYPE_12WORD = ButtonOption("20 words (128-bit entropy)")
+    TYPE_24WORD = ButtonOption("33 words (256-bit entropy)")
+
+    def __init__(self, k: int):
+        super().__init__()
+        self.k = k
+
+
+    def run(self):
+        button_data = [self.TYPE_12WORD, self.TYPE_24WORD]
+
+        selected_menu_num = self.run_screen(
+            ButtonListScreen,
+            title=_("Select SLIP-39 Word Count"),
```

Comment:
"Select" is making the title unnecessarily long and forcing it to scroll.

It might be more clear to use something like "Words Per Share"?